Can't compile lut3d.cl on MacOS / openCL fails as a result

**Describe the bug**
For some reason I can compile almost everything that is required for OpenCL to work except lut3d.cl.

**Platform (please complete the following information):**
 - OS: MacOS Mojave 10.14.4 with 5k display
 - Master 2.7.0 1154 g1ea1ab272

**To Reproduce**
Steps to reproduce the behavior:
1. I executed from terminal `/Applications/darktable.app/Contents/MacOS/darktable -d opencl`
2. pasted the output (end of it) below

**Output end, everything before has been compiled successfuly**
```
0.079294 [opencl_init] compiling program 'lut3d.cl' ..
0.079337 [opencl_fopen_stat] could not open file 
```'/Users/seb/.cache/darktable/cached_kernels_for_AMDRadeonPro580ComputeEngine_12Mar112019211443/lut3d.cl.bin'!
0.079358 [opencl_load_program] could not load cached binary program, trying to compile source
0.079379 [opencl_load_program] successfully loaded program from '/Applications/darktable.app/Contents/Resources/share/darktable/kernels/lut3d.cl'
0.157340 [opencl_build_program] could not build program: -11
0.157354 [opencl_build_program] BUILD STATUS: -2
0.157357 BUILD LOG:
0.157359 In file included from <program source>:21:
/Applications/darktable.app/Contents/Resources/share/darktable/kernels/common.h:34:1: warning: no previous prototype for function 'FC'
FC(const int row, const int col, const unsigned int filters)
^
/Applications/darktable.app/Contents/Resources/share/darktable/kernels/common.h:41:1: warning: no previous prototype for function 'FCxtrans'
FCxtrans(const int row, const int col, global const unsigned char (*const xtrans)[6])
^
Undeclared function '' called by function 'lut3d_tetrahedral'

0.157369 [opencl_init] failed to compile program 'lut3d.cl'!
0.157385 [opencl_init] FINALLY: opencl is NOT AVAILABLE on this system.
0.157389 [opencl_init] initial status of opencl enabled flag is OFF.

(darktable-bin:60251): GLib-GObject-WARNING **: 14:10:52.182: invalid cast from 'GtkMenuBar' to 'GtkWindow'

(darktable-bin:60251): Gtk-CRITICAL **: 14:10:52.182: gtk_window_add_accel_group: assertion 'GTK_IS_WINDOW (window)' failed

(darktable-bin:60251): Pango-WARNING **: 14:10:54.115: couldn't load font "sans-serif Condensed Not-Rotated 12.958984375", modified variant/weight/stretch as fallback, expect ugly output.

(darktable-bin:60251): GLib-GObject-CRITICAL **: 14:11:48.026: g_object_unref: assertion 'G_IS_OBJECT (object)' failed

I've just tried another 30 times. Always lut3d.cl still with 2 kind of failures :
1. The first one at the beginning of this issue : 
```
0.085669 [opencl_init] compiling program `lut3d.cl' ..
0.085712 [opencl_fopen_stat] could not open file `/Users/seb/.cache/darktable/cached_kernels_for_AMDRadeonPro580ComputeEngine_12Mar112019211443/lut3d.cl.bin'!
0.085717 [opencl_load_program] could not load cached binary program, trying to compile source
0.085739 [opencl_load_program] successfully loaded program from `/usr/local/share/darktable/kernels/lut3d.cl'
0.161545 [opencl_build_program] could not build program: -11
0.161558 [opencl_build_program] BUILD STATUS: -2
0.161561 BUILD LOG:
0.161563 In file included from <program source>:21:
/usr/local/share/darktable/kernels/common.h:34:1: warning: no previous prototype for function 'FC'
FC(const int row, const int col, const unsigned int filters)
^
/usr/local/share/darktable/kernels/common.h:41:1: warning: no previous prototype for function 'FCxtrans'
FCxtrans(const int row, const int col, global const unsigned char (*const xtrans)[6])
^
Undeclared function '' called by function 'lut3d_tetrahedral'

0.161573 [opencl_init] failed to compile program `lut3d.cl'!
0.161587 [opencl_init] FINALLY: opencl is NOT AVAILABLE on this system.
0.161589 [opencl_init] initial status of opencl enabled flag is OFF.
```
2.
```
0.087224 [opencl_init] compiling program `lut3d.cl' ..
0.087282 [opencl_fopen_stat] could not open file `/Users/seb/.cache/darktable/cached_kernels_for_AMDRadeonPro580ComputeEngine_12Mar112019211443/lut3d.cl.bin'!
0.087308 [opencl_load_program] could not load cached binary program, trying to compile source
0.087331 [opencl_load_program] successfully loaded program from `/usr/local/share/darktable/kernels/lut3d.cl'
10.260070 [opencl_build_program] could not build program: -11
10.260085 [opencl_build_program] BUILD STATUS: -2
10.260090 BUILD LOG:
10.260092 Error returned by cvms_element_build_from_source
10.260096 [opencl_init] failed to compile program `lut3d.cl'!
10.260115 [opencl_init] FINALLY: opencl is NOT AVAILABLE on this system.
10.260119 [opencl_init] initial status of opencl enabled flag is OFF.
```

ok, i compared the darktable start cleaning all stuff first:
within ~/src/darktable/build
```rm -rf *
rm -rf ~/.cache/darktable/cached_kernels*
sudo rm -rf /usr/local/share/darktable
sudo rm -rf /usr/local/lib/darktable
cmake ... as in Build.txt 
make
sudo make install
```

the difference between a successful build of a *.cl file  an a failing build is:
successful:
successful (starting from /usr/local/bin/darktable):
```
4.672414 [opencl_init] compiling program `bilateral.cl' ..
4.672509 [opencl_fopen_stat] could not open file `/Users/martinstraeten/.cache/darktable/cached_kernels_for_IrisPro_12May32019174717/bilateral.cl.bin'!
4.672519 [opencl_load_program] could not load cached binary program, trying to compile source
4.672537 [opencl_load_program] successfully loaded program from `/usr/local/share/darktable/kernels/bilateral.cl'
4.782418 [opencl_build_program] successfully built program
4.782452 [opencl_build_program] BUILD STATUS: 0
4.782458 BUILD LOG:
4.782461 In file included from <program source>:21:
/usr/local/share/darktable/kernels/common.h:34:1: warning: no previous prototype for function 'FC'
```

failing (starting from darktable.app):
```
0.808088 [opencl_init] compiling program `bilateral.cl' ..
0.808170 [opencl_fopen_stat] could not open file `/Users/martinstraeten/.cache/darktable/cached_kernels_for_IrisPro_12May32019174717/bilateral.cl.bin'!
0.808178 [opencl_load_program] could not load cached binary program, trying to compile source
0.808197 [opencl_load_program] successfully loaded program from `/Applications/darktable.app/Contents/Resources/share/darktable/kernels/bilateral.cl'
0.816268 [opencl_build_program] could not build program: -11
0.816283 [opencl_build_program] BUILD STATUS: -2
0.816285 BUILD LOG:
0.816287 <program source>:21:10: fatal error: 'common.h' file not found
#include "common.h"
```
one start later common.h is found at this step also for darktable.app but then not found for a later *.cl

i never got the line
`Error returned by cvms_element_build_from_source`
but I have a good old Nvidia inside an no AMD ATI ;) googling for this error gave me: https://discussions.apple.com/thread/6724078
--> seems to be gpu specific. maybe there are some hints in https://darktable.gitlab.io/doc/en/darktable_and_opencl_amd.html

--- a/data/kernels/lut3d.cl
+++ b/data/kernels/lut3d.cl
@@ -36,7 +36,7 @@ lut3d_tetrahedral(read_only image2d_t in, write_only image2d_t out, const int wi
   input = clamp(input, (float4)0.0f, (float4)1.0f);
 
   rgbd = input * (float)(level - 1);
-  rgbi = min( max( convert_int4(rgbd), 0), (int)(level - 2));
+  rgbi = min( max( convert_int4(rgbd), (int4)0), (int4)(level - 2));
 
 // delta r, g, b
   rgbd = rgbd - convert_float4(rgbi);
@@ -116,7 +116,7 @@ lut3d_trilinear(read_only image2d_t in, write_only image2d_t out, const int widt
   input = clamp(input, (float4)0.0f, (float4)1.0f);
 
   rgbd = input * (float)(level - 1);
-  rgbi = min( max( convert_int4(rgbd), 0), (int)(level - 2));
+  rgbi = min( max( convert_int4(rgbd), (int4)0), (int4)(level - 2));
 
   // delta r, g, b
   rgbd = rgbd - convert_float4(rgbi);
@@ -172,7 +172,7 @@ lut3d_pyramid(read_only image2d_t in, write_only image2d_t out, const int width,
   input = clamp(input, (float4)0.0f, (float4)1.0f);
 
   rgbd = input * (float)(level - 1);
-  rgbi = min( max( convert_int4(rgbd), 0), (int)(level - 2));
+  rgbi = min( max( convert_int4(rgbd), (int4)0), (int4)(level - 2));
 
   // delta r, g, b
   rgbd = rgbd - convert_float4(rgbi);

--- a/data/kernels/lut3d.cl
+++ b/data/kernels/lut3d.cl
@@ -34,12 +34,12 @@ lut3d_tetrahedral(read_only image2d_t in, write_only image2d_t out, const int wi
   float4 output = (float4)(0.0f);
 
   input = clamp(input, (float4)0.0f, (float4)1.0f);
-  
-  rgbd = input * (float)(level - 1); 
+
+  rgbd = input * (float)(level - 1);
   rgbi = min( max( convert_int4(rgbd), 0), (int)(level - 2));
 
 // delta r, g, b
-  rgbd = rgbd - convert_float4(rgbi); 
+  rgbd = rgbd - convert_float4(rgbi);
 
 // indexes of P000 to P111 in clut
   const int color = rgbi.x + rgbi.y * level + rgbi.z * level2;
@@ -107,7 +107,7 @@ lut3d_trilinear(read_only image2d_t in, write_only image2d_t out, const int widt
   const int y = get_global_id(1);
   float4 tmp1;
   float4 tmp2;
-  
+
   if(x >= width || y >= height) return;
 
   float4 input = read_imagef(in, sampleri, (int2)(x, y));
@@ -115,11 +115,11 @@ lut3d_trilinear(read_only image2d_t in, write_only image2d_t out, const int widt
 
   input = clamp(input, (float4)0.0f, (float4)1.0f);
 
-  rgbd = input * (float)(level - 1); 
+  rgbd = input * (float)(level - 1);
   rgbi = min( max( convert_int4(rgbd), 0), (int)(level - 2));
 
   // delta r, g, b
-  rgbd = rgbd - convert_float4(rgbi); 
+  rgbd = rgbd - convert_float4(rgbi);
 
   // indexes of P000 to P111 in clut
   const int color = rgbi.x + rgbi.y * level + rgbi.z * level2;
@@ -141,7 +141,7 @@ lut3d_trilinear(read_only image2d_t in, write_only image2d_t out, const int widt
   const float4 clut101 = (float4)(clut[i101], clut[i101+1], clut[i101+2], 0.0f);
   const float4 clut011 = (float4)(clut[i011], clut[i011+1], clut[i011+2], 0.0f);
   const float4 clut111 = (float4)(clut[i111], clut[i111+1], clut[i111+2], 0.0f);
-  
+
   tmp1 = clut000*(1.0f-rgbd.x) + clut100*rgbd.x;
   tmp2 = clut010*(1.0f-rgbd.x) + clut110*rgbd.x;
   output = tmp1*(1.0f-rgbd.y) + tmp2*rgbd.y;
@@ -171,11 +171,11 @@ lut3d_pyramid(read_only image2d_t in, write_only image2d_t out, const int width,
 
   input = clamp(input, (float4)0.0f, (float4)1.0f);
 
-  rgbd = input * (float)(level - 1); 
+  rgbd = input * (float)(level - 1);
   rgbi = min( max( convert_int4(rgbd), 0), (int)(level - 2));
 
   // delta r, g, b
-  rgbd = rgbd - convert_float4(rgbi); 
+  rgbd = rgbd - convert_float4(rgbi);
 
   // indexes of P000 to P111 in clut
   const int color = rgbi.x + rgbi.y * level + rgbi.z * level2;
@@ -226,7 +226,6 @@ lut3d_none(read_only image2d_t in, write_only image2d_t out, const int width, co
   if(x >= width || y >= height) return;
 
   float4 input = read_imagef(in, sampleri, (int2)(x, y));
-  
+
   write_imagef(out, (int2)(x, y), input);
 }
-

